### PR TITLE
supported gpt-oss

### PR DESCRIPTION
**Pull request by FrankLeeeee**
## Motivation

Added support for GPT-OSS.

## Modifications

1. Added abstraction for conversation parsing
2. implemented GPT-OSS TP version

## Related Issues

N/A

## Accuracy Test

N/A

## Benchmark & Profiling

N/A

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [x] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://sgl-fru7574.slack.com/archives/C09784E3EN6 to discuss your PR.

### CODE REVIEW

File: scripts/prepare_data.py [hunk 1]
```diff
@@ -231,9 +236,17 @@ def main():
...
+        ds = load_dataset("shuaills/perfect-blend-gptoss-20B")["train"]
+        ds = ds.map(add_index, with_indices=True)
+        proc_fn = process_sharegpt_row
     else:
         raise ValueError(
+<<<<<<< HEAD
             "This script only supports ultrachat_200k and sharegpt datasets for demo purpose, if you wish to use other datasets, please modify this script."
+=======
+            f"This script only supports sharegpt, ultrachat, opc, and perfect-blend-gptoss-20B datasets for demo purpose, if you wish to use other datasets, please modify this script."
+>>>>>>> a35701e (fixed gptoss chat template)
```
[the remark this answers is not in the document]
Comment (FrankLeeeee) [reply scripts/prepare_data.py]: resolved.

File: examples/run_gpt_oss_20b_eagle3_online.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# train eagle3 for llama3.1-8b
+NUM_GPUS=${1:-8}
+
+torchrun \
+    --standalone \
+    --nproc_per_node $NUM_GPUS \
+    $ROOT_DIR/scripts/train_eagle3_online.py \
+    --target-model-path openai/gpt-oss-20b \
+    --draft-model-config $ROOT_DIR/configs/gpt-oss-20B-eagle3.json \
+    --train-data-path /data/shenggui/data/perfect-data-temp.jsonl \
```
[the remark this answers is not in the document]
Comment (FrankLeeeee) [reply examples/run_gpt_oss_20b_eagle3_online.sh]: Resolved.

File: examples/run_gpt_oss_120b_eagle3_online.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+ROOT_DIR=$(dirname $SCRIPT_DIR)
+
+# train eagle3 for llama3.1-8b
```
[the remark this answers is not in the document]
Comment (FrankLeeeee) [reply examples/run_gpt_oss_120b_eagle3_online.sh]: Resolved.

File: examples/run_gpt_oss_20b_eagle3_online.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+ROOT_DIR=$(dirname $SCRIPT_DIR)
+
+# train eagle3 for llama3.1-8b
```
[the remark this answers is not in the document]
Comment (FrankLeeeee) [reply examples/run_gpt_oss_20b_eagle3_online.sh]: Fixed.

File: specforge/data/preprocessing.py [hunk 5]
```diff
@@ -137,6 +138,13 @@ def preprocess_conversations(
     # prepare result
     results = {"input_ids": [], "loss_mask": [], "attention_mask": []}
 
+    if chat_template.parser_type == "general":
+        parser = GeneralParser(tokenizer, chat_template)
+    elif chat_template.parser_type == "openai-harmony":
+        parser = HarmonyParser(tokenizer, chat_template)
+    else:
+        raise ValueError(f"Invalid parser type: {chat_template.parser_type}")
```
[the remark this answers is not in the document]
Comment (FrankLeeeee) [reply specforge/data/preprocessing.py:146]: fixed.
